hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/common/AbstractWriteFunction.java

```diff
@@ -45,4 +92,46 @@
    * @param event The event
    */
   public abstract void handleOperatorEvent(OperatorEvent event);
+
+  /**
+   *  Extract TimeStamp from input value with specified event time field.
+   * @param value The input value
+   * @return the new timestamp for current.
+   */
+  public  long extractTimestamp(I value) {
+    if (value instanceof HoodieAvroRecord) {
+      return extractTimestamp((HoodieAvroRecord) value);
+    }
+    return extractTimestamp((RowData) value);
+  }
```

Comment:
Can we move these logic into `AbstractStreamWriteFunction.java`.

Reply:
The reason why put it here is BulkInsertWriteFunction only extends BulkInsertWriteFunction instead of AbstractStreamWriteFunction, if moving this code to AbstractStreamWriteFunction will make a specified way to deal with BulkInsertWriteFunction